pkg/user/repository.go

```diff
@@ -30,6 +30,17 @@ func (r repository) create(u *model.User) error {
 	return err
 }
 
+func (r repository) findAll() ([]*model.User, error) {
+	var u []*model.User
+
+	err := r.db.
+		Preload("Groups").
+		Preload("AdminGroups").
+		Find(&u).Error
+
+	return u, err
```

Comment:
I think we should not return the gorm error but something like `fmt.Errorf("failed to find all users: %v", err)` in case of an error. I don't think we agreed on something yet so we should probably do that to return consistent errors.

Like what phrase to use `failed to...` and whether to return the original error to the user (like with `%v` or not and log the details only for us).

Reply:
Okay, that makes sense. Did you mean something like:
```
func (r repository) findAll() ([]*model.User, error) {
	var u []*model.User

	err := r.db.
		Preload("Groups").
		Preload("AdminGroups").
		Find(&u).Error
	if err != nil {
		log.Printf("failed to find all users: %v", err)
		return nil, fmt.Errorf("failed to find all users: %v", err)
	}

	return u, err
}
```

🤔

Reply:
yes 😄 the last return can then be `return users, nil`